### PR TITLE
Revise "AOA multiplexing similar to Android BT/SPP multiplexing" proposal

### PR DESCRIPTION
**Pull request by shiniwat**
This proposal aims to add AOA multiplexing capability to SdlRouterService.

### CODE REVIEW

**Comment by shiniwat** [comment]
Hi @theresalech , this pull-request is to update "SDL 0095 AOA multiplexing similar to Android BT/SPP multiplexing" proposal. This is ready for review. Thanks.

File: proposals/0095-AOA-multiplexing.md [hunk 1]
```diff
@@ -1,72 +1,193 @@
...
 
 ## Motivation
 
-Currently, only one AOA app can connect to SDL core because of AOA limitation. However, a few navigation apps will be released. And, video projection functionality will be opened for not only navigation but also other category apps. And not projection but using wide band-width and stable transport apps (ex. using high quality audio streaming or pass thru), may be considered in the future. This proposal is for supporting and realizing these requirements.
+Currently SdlRouterService works for MultiplexTransport, which supports Bluetooth transport as the underlying transport. This proposal extends SdlRouterService to support both Bluetooth and AOA (Android Open Accessory) transports, so that multiple SDL applications can share the AOA transport without having to worry about USB connection.
+AOA protocol basically allows only single application to access the USB accessory -- if multiple applications are designed to use the same accessory, Android system asks which application to use the accessory and shows a permission dialog. This proposal allows multiple SDL applications to share the USB accessory.
```
Comment (theresalech) [inline proposals/0095-AOA-multiplexing.md]: Please update first sentence to "AOA protocol basically allows only a single application to access... if multiple applications are designed to use the same accessory, the Android system asks which application should use the accessory..."

File: proposals/0095-AOA-multiplexing.md [hunk 2]
```diff
@@ -1,72 +1,193 @@
...
-- Case: "old apps" only: Only one of old apps can connect.
-- Case: "new apps, no old apps": normal case, all apps can connect.
+This feature introduces TransportType for TransportBroker and SdlRouterService. While this approach should have no obvious downsides, backward compatibility should be taken into account as much as we can.
+In particular, the following cases need to be confirmed:
+- Case #1: Older proxy's TransportBroker binds to new SdlRouterService. In this case, SdlRouterService assumes TransportType == MULTIPLEX for backward compatibility.
+- Case #2: Newer proxy's TransportBroker binds to older SdlRouterService. The older SdlRouterService won't support AOA multiplexing. In this case, the expected behavior is "don't bind to older SdlRouterService; instead, start and bind to newer (local) SdlRouterService". We can utilize existing version check and trusted router logic to make this case to work.
```
Comment (theresalech) [inline proposals/0095-AOA-multiplexing.md]: Please update the last sentence to "We can utilize existing version check and trusted router logic to make this case work."

File: proposals/0095-AOA-multiplexing.md [hunk 3]
```diff
@@ -1,72 +1,193 @@
...
 
+This feature also increases the IPC transaction between TransportBroker and SdlRouterService. While Android system has Binder Transaction Limit, which is explained at [TransactionTooLargeException Android document](https://developer.android.com/reference/android/os/TransactionTooLargeException.html), we won’t run into TransactionTooLargeException cases in real scenario based on our test, unless underlying transport has a fatal error. The fatal error case would be, for example, the case where we cannot write Bluetooth socket and/or USB's ParcelFileDescriptor for some reason. Those fatal error cases can be discussed in outside of this proposal.
```
Comment (theresalech) [inline proposals/0095-AOA-multiplexing.md]: Please update last sentence to "Those fatal error cases can be discussed outside of this proposal."

File: proposals/0095-AOA-multiplexing.md [hunk 4]
```diff
@@ -1,72 +1,193 @@
...
 
 ## Impact on existing code
 
-New service and TransportConfig will be added. These modification is not big impact to existing code.
+All the change will be made to Android Proxy, and there's no change required to SDL Core and/or other components.
```
Comment (theresalech) [inline proposals/0095-AOA-multiplexing.md]: Please update to "All changes will be made..."

**Comment by shiniwat** [comment]
Hi @theresalech, many thanks for the review. I have updated the proposal in accordance with your comments. Thanks!

**Pull request merged**